authorization/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	authorizationv1 "github.com/openshift/api/authorization/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AccessRestriction is used to guard specific actions without invasive changes to the cluster's default RBAC policy.
+// It supports a required blacklist based restriction with an optional whitelist opt-out.
+// It never grants any privileges - it can only be used to take privileges away.
+type AccessRestriction struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec defines when this restriction is imposed and how to satisfy it.
+	Spec AccessRestrictionSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// AccessRestrictionSpec holds the matching requirements.
+// MatchAttributes and DeniedSubjects (blacklist) are required.
+// AllowedSubjects (whitelist) may optionally be specified.
+type AccessRestrictionSpec struct {
+	// If these rules cover the current request, then this restriction applies.
+	// If AllowedSubjects is set, those subjects can perform the matching actions even if they exist in DeniedSubjects.
+	// The subjects in DeniedSubjects are restricted from performing the matching actions unless they exist in AllowedSubjects.
+	// If the subject does not exist in either AllowedSubjects or DeniedSubjects, the request is allowed.
```

Comment:
", then this authorizer has no opinion on this request.  "

Reply:
Updated.